akka-stream/src/main/scala/akka/stream/impl/TraversalBuilder.scala

```diff
@@ -156,23 +156,26 @@ object TraversalBuilder {
   /**
    * Create a generic traversal builder starting from an atomic module.
    */
-  def atomic(module: AtomicModule[Shape, Any], attributes: Attributes = Attributes.none): TraversalBuilder = {
+  def atomic(module: AtomicModule[Shape, Any], attributes: Attributes): TraversalBuilder = {
     initShape(module.shape)
 
-    if (module.shape.outlets.isEmpty) {
-      val b = CompletedTraversalBuilder(
-        traversalSoFar = MaterializeAtomic(module, Array.ofDim[Int](module.shape.outlets.size)),
-        inSlots = module.shape.inlets.size,
-        inToOffset = module.shape.inlets.map(in ⇒ in → in.id).toMap,
-        attributes)
-      b
-    } else {
-      AtomicTraversalBuilder(
-        module,
-        Array.ofDim[Int](module.shape.outlets.size),
-        module.shape.outlets.size,
-        attributes)
-    }
+    val builder =
+      if (module.shape.outlets.isEmpty) {
+        val b = CompletedTraversalBuilder(
+          traversalSoFar = MaterializeAtomic(module, Array.ofDim[Int](module.shape.outlets.size)),
+          inSlots = module.shape.inlets.size,
+          inToOffset = module.shape.inlets.map(in ⇒ in → in.id).toMap,
+          Attributes.none)
+        b
+      } else {
+        AtomicTraversalBuilder(
+          module,
+          Array.ofDim[Int](module.shape.outlets.size),
+          module.shape.outlets.size,
+          Attributes.none)
+      }
+    // important to use setAttributes because it will create island for async (dispatcher attribute)
+    builder.setAttributes(attributes)
```

Comment:
The CCE is because setAttributes changes the `islandTag` to `GraphStageTag`.
A hack is
```scala
    val x = builder.setAttributes(attributes).copy(islandTag = builder.islandTag)
    println(s"# xxx $x vs $builder") // FIXME
    x
```